system/lib/pthread/proxying.c

```diff
@@ -642,3 +651,34 @@ double _emscripten_run_on_main_thread_js(int func_index,
   }
   return 0;
 }
+
+double _emscripten_await_on_main_thread_js(int func_index,
```

Comment:
I do think there is enough code shared with `_emscripten_run_on_main_thread_js` that it may make sense to combine them, but I don't feel strongly about it. If you do keep them separate, you should explicitly initialize the `.ctx` field to `NULL` in `_emscripten_run_on_main_thread_js` to make sure it is initialized .